x/sync/syncworkheap.go

```diff
@@ -128,61 +135,64 @@ func (wh *syncWorkHeap) MergeInsert(item *syncWorkItem) {
 		wh.remove(mergedAfter)
 		// update the priority
 		mergedBefore.workItem.priority = math.Max(mergedBefore.workItem.priority, mergedAfter.workItem.priority)
-		heap.Fix(wh, mergedBefore.heapIndex)
+		heap.Fix(&wh.innerHeap, mergedBefore.heapIndex)
 	}
 
 	// nothing was merged, so add new item to the heap
 	if mergedBefore == nil && mergedAfter == nil {
 		// We didn't merge [item] with an existing one; put it in the heap.
-		heap.Push(wh, &heapItem{workItem: item})
+		wh.Insert(item)
 	}
 }
 
 // Deletes [item] from the heap.
 func (wh *syncWorkHeap) remove(item *heapItem) {
 	oldIndex := item.heapIndex
-	newLength := len(wh.priorityHeap) - 1
+	newLength := len(wh.innerHeap) - 1
 
 	// swap with last item, delete item, then fix heap if required
-	wh.Swap(newLength, item.heapIndex)
-	wh.priorityHeap[newLength] = nil
-	wh.priorityHeap = wh.priorityHeap[:newLength]
+	wh.innerHeap.Swap(newLength, item.heapIndex)
+	wh.innerHeap[newLength] = nil
+	wh.innerHeap = wh.innerHeap[:newLength]
 
 	// the item was already the last item, so nothing needs to be fixed
 	if oldIndex != newLength {
-		heap.Fix(wh, oldIndex)
+		heap.Fix(&wh.innerHeap, oldIndex)
 	}
```

Comment:
We should be using: `heap.Remove` here https://pkg.go.dev/container/heap#Remove.